pycode/test/test_get_vaccination_data.py

```diff
@@ -0,0 +1,284 @@
+
+    @patch('epidemiology.epidata.getVaccinationData.pd.read_csv',
+           side_effect=ImportError())
+    def test_download_not_working(self, mock_download):
+        df = gvd.download_vaccination_data()
+        self.assertTrue(df.empty, "Vaccination Data is empty.")
+
+    def test_split_column_based_on_values(self):
+        col_names_vacc_data = [
+            'Impfdatum', 'LandkreisId_Impfort', 'Altersgruppe', 'Impfschutz',
+            'Anzahl']
+
+        vacc_data = [
+            ('2020-12-27', str(10001), '12-17', 1, 10),
+            ('2020-12-27', str(10001), '12-17', 2, 15),
+            ('2020-12-27', str(10001), '12-17', 3, 72),
+            ('2020-12-27', str(10001), '18-59', 1, 2),
+            ('2020-12-27', str(10001), '18-59', 2, 3),
+            ('2020-12-27', str(10001), '18-59', 3, 222),
+            ('2020-12-27', str(10001), '60+', 1, 22),
+            ('2020-12-27', str(10001), '60+', 2, 332),
+            ('2020-12-27', str(10001), '60+', 3, 76)
+        ]
+        df_to_split = pd.DataFrame(
+            vacc_data, columns=col_names_vacc_data)
+
+        column_ident = 'Impfschutz'
+        column_vals_name = 'Anzahl'
+        new_col_labels = ['Vacc_partially', 'Vacc_completed', 'Vacc_refreshed']
+        df_split = gvd.split_column_based_on_values(
+            df_to_split, column_ident, column_vals_name, new_col_labels)
+
+        new_column_names1 = [
+            'Impfdatum',
+            'LandkreisId_Impfort',
+            'Altersgruppe',
+            'Vacc_partially']
+        new_column_names2 = [
+            'Impfdatum',
+            'LandkreisId_Impfort',
+            'Altersgruppe',
+            'Vacc_completed']
+        new_column_names3 = [
+            'Impfdatum',
+            'LandkreisId_Impfort',
+            'Altersgruppe',
+            'Vacc_refreshed']
+        new_column_names = []
+        new_column_names.append(new_column_names1)
+        new_column_names.append(new_column_names2)
+        new_column_names.append(new_column_names3)
+
+        indent_unique = df_to_split[column_ident].unique()
+        number_unique_idents_df = []
+        for i in range(0, len(indent_unique)):
+            number_unique_idents_df.append(
+                len(df_to_split[df_to_split[column_ident] == indent_unique[i]]))
+
+        self.assertEqual(
+            len(df_split),
+            3, 'Did not divide dataframe in three sperate ones.')
+        for i in range(0, len(new_col_labels)):
+            self.assertEqual(
+                number_unique_idents_df[i],
+                len(df_split[i]),
+                'Length of Dataframe is wrong.')
+            column_names_i = df_split[i].columns
+            for j in range(0, len(column_names_i)):
+                self.assertEqual(
+                    new_column_names[i][j],
+                    column_names_i[j],
+                    'Column names do not match.')
+
+    def test_split_column_based_on_values_ident_length_wrong(self):
```

Comment:
What do you actually test here? There is no change in the assertion or equal tests below? Is this a good test?

Reply:
Which test do you mean exactly?

Reply:
If you mean the Asserts:

This only tests if it got split 3 times and if the column names are correct, as well as that there are the right amount of rows. The test does not test if the entries are correct, which it maybe should.

Reply:
ok